threedi_models_and_simulations/workers.py

```diff
@@ -83,6 +83,8 @@ def run(self):
                 }
                 for status in finished_simulations_statuses
             }
+            logger.debug(f"Fetched {len(finished_simulations_data)} finished simulation statuses")
+            time.sleep(1)
             self.simulation_finished.emit(finished_simulations_data)
```

Comment:
This dict here is for sure filled at this point - also without the `sleep()`.

There is a race condition somewhere else, probably?

Reply:
I didn't found any place in the code which my cause a race condition. `simulation_finished` here is emitted even before establishing a connection to the websocket with running simulations progresses. I can dig, more to find an explanation of that behaviour, but for a time beign I would treat it as a working workaround.

Reply:
Probably the signal is emitted too early. That explains why it happens when there are only few results. Also why it is coming up now because I think @jpprins1 improved performance.

Could it be that the listener is just not there yet?